src/components/cursor.js

```diff
@@ -475,19 +475,25 @@ module.exports.Component = registerComponent('cursor', {
   /**
    * Helper to emit on both the cursor and the intersected entity (if exists).
    */
-  twoWayEmit: function (evtName) {
+  twoWayEmit: function (evtName, originalEvent) {
     var el = this.el;
     var intersectedEl = this.intersectedEl;
     var intersection;
 
     intersection = this.el.components.raycaster.getIntersection(intersectedEl);
     this.eventDetail.intersectedEl = intersectedEl;
     this.eventDetail.intersection = intersection;
+    if (originalEvent) {
+      this.eventDetail.originalEvent = originalEvent;
+    }
     el.emit(evtName, this.eventDetail);
 
     if (!intersectedEl) { return; }
 
     this.intersectedEventDetail.intersection = intersection;
+    if (originalEvent) {
+      this.intersectedEventDetail.originalEvent = originalEvent;
```

Comment:
maybe calling `touchEvent` or `mouseEvent` depending on the case instead of generic `originalEvent` is more clear. Uses the Web standard nomenclature

Reply:
Yes, I like that suggestion.  I was aiming for minimal change to the interface, but I agree this will be more usable.  Application developers need to consider whether hey are reading a mouse event or a touch event as they have different properties, and this will help steer them in the correct direction.

PR updated with this change.